rmc/utils/constraint.py

```diff
@@ -192,54 +195,64 @@ def calculate_exp_from_mu(
         - Assumes that input Table is filtered to autosomes/PAR only, X nonPAR only, or Y nonPAR only.
         - Assumes that input Table contains coverage and plateau models in its global annotations
             (`coverage_model`, `plateau_models`).
-        - Adds `expected` and `coverage_correction` annotations.
+        - Adds `expected` annotation.
 
     :param context_ht: Variant-level input context Table.
     :param locus_type: Locus type of input Table. One of "X", "Y", or "autosomes".
         NOTE: Will treat any input other than "X" or "Y" as autosomes.
     :param groupings: List of Table fields used to group Table to adjust mutation rate.
         Table must be annotated with these fields. Default is `GROUPINGS`.
-    :return: Table annotated with per-variant expected counts and coverage correction.
+    :param overwrite: Whether to overwrite temporary data. Default is False.
+    :param n_partitions: Number of desired partitions for temporary grouped context Table. Default is 5000.
+    :return: Table annotated with per-variant expected counts.
     """
-    logger.info(
-        "Grouping by %s and aggregating mutation rates within groupings...", groupings
-    )
-    group_ht = context_ht.group_by(*groupings).aggregate(
-        mu_agg=hl.agg.sum(context_ht.mu_snp),
-        n_grouping=hl.agg.count(),
-        # `cpg` is a function of `context`, `ref``, `alt` which are part of `groupings`
-        # so will be the same for all alleles in a grouping
-        cpg=hl.agg.take(context_ht.cpg, 1)[0],
+    mu_expr = context_ht.mu_snp
+    cov_corr_expr = get_coverage_correction_expr(
+        context_ht.coverage, context_ht.coverage_model
     )
 
-    logger.info("Adjusting aggregated mutation rates with plateau model...")
     if locus_type == "X":
-        model = group_ht.plateau_x_models["total"][group_ht.cpg]
+        plateau_model = context_ht.plateau_x_models
     elif locus_type == "Y":
-        model = group_ht.plateau_y_models["total"][group_ht.cpg]
+        plateau_model = context_ht.plateau_y_models
     else:
-        model = group_ht.plateau_models["total"][group_ht.cpg]
-
-    group_ht = group_ht.annotate(mu_adj=group_ht.mu_agg * model[1] + model[0])
+        plateau_model = context_ht.plateau_models
+
+    agg_expr = {
+        "mu": hl.agg.sum(mu_expr * cov_corr_expr),
+        "mu_agg": hl.agg.sum(mu_expr),
+    }
```

Comment:
Is there a reason this is here? It doesn't get used downstream, only the expected is annotated (unless it's to keep note of in the case where we need it if the model changes)

Reply:
mostly just so we have it in case we need it (to troubleshoot)